packages/delivery-service/src/index.ts

```diff
@@ -0,0 +1,104 @@
+import { Axios } from 'axios';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import express from 'express';
+import http from 'http';
+import path from 'path';
+import { Server } from 'socket.io';
+import winston from 'winston';
+import { startCleanUpPendingMessagesJob } from './cleanup/cleanUpPendingMessages';
+import { getDeliveryServiceProperties } from './config/getDeliveryServiceProperties';
+import Delivery from './delivery';
+import { onConnection } from './messaging';
+import { getDatabase } from './persistence/getDatabase';
+import Profile from './profile';
+import RpcProxy from './rpc/rpc-proxy';
+import { logInfo } from '@dm3-org/dm3-lib-shared';
+import 'dotenv/config';
+
+import {
+    errorHandler,
+    getWeb3Provider,
+    logError,
+    logRequest,
+    readKeysFromEnv,
+    socketAuth,
+} from './utils';
+import Notifications from './notifications';
+
+const app = express();
+app.use(express.json({ limit: '50mb' }));
+app.use(express.urlencoded({ limit: '50mb' }));
+
+const server = http.createServer(app);
+
+//TODO remove
+app.use(cors());
+app.use(bodyParser.json());
+
+declare global {
+    var logger: winston.Logger;
+}
+
+global.logger = winston.createLogger({
+    level: process.env.LOG_LEVEL ?? 'info',
+    transports: [new winston.transports.Console()],
+});
+
+(async () => {
+    const io = new Server(server, {
+        cors: {
+            origin: '*',
+            methods: 'GET,HEAD,PUT,PATCH,POST,DELETE',
+            preflightContinue: false,
+            optionsSuccessStatus: 204,
+        },
+    });
+
+    app.locals.io = io;
+
+    app.locals.keys = readKeysFromEnv(process.env);
+    app.locals.deliveryServiceProperties = getDeliveryServiceProperties();
+
+    app.locals.db = await getDatabase();
+    app.locals.web3Provider = getWeb3Provider(process.env);
+
+    app.use(logRequest);
+
+    app.get('/hello', (req, res) => {
+        return res.send('Hello DM3');
+    });
+
+    /**
+     *     needed
+     */
+    app.use('/profile', Profile()); // remove in backend
+    app.use('/delivery', Delivery()); // remove in backend
+    app.use(
+        '/notifications',
+        Notifications(app.locals.deliveryServiceProperties),
+    ); // remove in backend
+    app.use(logError);
+    app.use(errorHandler);
+    //@ts-ignore
```

Comment:
Why is ts-ignore used here?

Reply:
Not sure, removing it now.

Reply:
These [functions](https://github.com/dm3-org/dm3/blob/54347b4af7156bf5f34e5120d1eef747c0d692f4/packages/delivery-service/src/messaging.ts#L20) expect `app` to implement the `express` and `WithLocals` interfaces. They use `app.locals.[...]`. Any tips on how to explicitly check for these interfaces without bloating too much?

Reply:
The withLocals interface attempts to specify the structure of the req.app.locals object that otherwise would be empty. The main advantage here is that the compiler would know the function signatures of the local vats, especially the IDatebase var. 
To tell the compiler that you've assigned all the mandatory locals file between 58 and 64 use 
```app as express.Express & { locals: WithLocals["locals"] })```

Reply:
thx